### PR TITLE
Download list

### PR DESCRIPTION
**Pull request by davidcmoulton**
Builds a list of related items to download, and services to use with the current article's content. Analogous to the non-js presentation of the downloads list on an eLife article.

### CODE REVIEW

**Comment by davidcmoulton** [comment]
@alex-ketch Since merging 44c0342 none of the eLife DOM changes in https://github.com/stencila/thema/pull/111/files#diff-f28cb2a39f1904fe2a99f06f90e7f9d5 are manifesting in the built page: neither the download menu that's being introduced here, nor the date formatting which was already in `next`.

Actually, I've just checked `next` and the [date formatting](https://github.com/stencila/thema/blob/next/src/themes/elife/index.ts#L18)'s not happening there either now. I've had a look but can't see what's wrong here. Any pointers gratefully received! 😅

**Comment by alex-ketch** [comment]
Hi @davidcmoulton, took a look and it seems that the issue was due to [changes made in Encoda ](https://github.com/stencila/thema/pull/106#issuecomment-599032506).
Apologies for not making the required changes explicit.
The cause is that the due to a change in date encoding in Encoda, the selector being used for `dateEl` should be changed from `first(':--Date')` to `first(':--datePublished)`.

https://github.com/stencila/thema/blob/b35092ff1a7f7a575f22e0521389376fac3abe66/src/themes/elife/index.ts#L16

That should now find the element, and avoid the early termination on line 17, and generating the download links section.
https://github.com/stencila/thema/blob/b35092ff1a7f7a575f22e0521389376fac3abe66/src/themes/elife/index.ts#L17

Let me know if there's anything else, and I apologize for the miscommunication.

**Comment by davidcmoulton** [comment]
@alex-ketch Thanks very much, that's fixed it 👍 Do you think Encoda changes like this are likely to occur for a while, or was that an occasional thing? Trying to gauge the stability of things I'm building the theme against.

**Comment by nokome** [comment]
Hi @davidcmoulton. There are unlikely to be breaking changes to the HTML encodings for at least two or three months. Any changes are likely to be adding additional information. But please do let us know if you think that there are changes that you think should be made earlier i.e. if there are better ways to encode certain information.

File: src/themes/elife/downloads.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    case 'bibtex':
+      return `https://elifesciences.org/articles/${id}.bib`
+    case 'ris':
+      return `https://elifesciences.org/articles/${id}.ris`
+    case 'mendeley':
+      return `https://www.mendeley.com/import?doi=10.7554/eLife.${id}`
+    case 'readcube':
+      return `https://www.readcube.com/articles/10.7554/eLife.${id}`
+    case 'papers':
+      return `papers2://url/https%3A%2F%2Felifesciences.org%2Farticles%2F46206?title=${encodeURIComponent(
```
Comment (giorgiosironi) [inline src/themes/elife/downloads.ts:14]: what's the magic number `46206`?

Comment (davidcmoulton) [reply src/themes/elife/downloads.ts:14]: 👍 Good spot, this should be `${id}`.

File: src/themes/elife/test/eifeDataProvider.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    test.todo('it exposes the URI of the article PDF')
+
+    test.todo('it exposes the URI of the figures PDF')
+  })
+
+  describe('being given an invalid article id', () => {
+    it('throws an ReferenceError', async () => {
+      const fetchMock = (): Promise<ArticleData> =>
+        Promise.resolve({ status: 404 })
```
Comment (davidcmoulton) [inline src/themes/elife/test/eifeDataProvider.test.ts:31]: Would be better use`fetch`'s `Repsonse.ok` property to determine if there's a problem & react accordingly.

File: src/themes/elife/test/eifeDataProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      await expect(
+        dataProvider.query('invalidArticleId', fetchMock)
+      ).rejects.toThrow(
+        new ReferenceError('Invalid eLife article id: invalidArticleId')
```
Comment (davidcmoulton) [inline src/themes/elife/test/eifeDataProvider.test.ts:35]: Too specific. The fetch could fail for many reasons.

File: src/themes/elife/test/eifeDataProvider.test.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      ).resolves.not.toThrow()
+    })
+
+    it('it exposes the URI of the article PDF', async () => {
+      const fetchMock = (): Promise<Response> =>
+        Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve({ pdf: 'path-to-the.pdf' })
+        })
+      await expect(dataProvider.query('someId', fetchMock)).resolves.toEqual({
```
Comment (davidcmoulton) [inline src/themes/elife/test/eifeDataProvider.test.ts:24]: Tried using `toContainEqual` here in order to the check for only the piece of data we care about in this specific test, but that kept failing unexpectedly. Also at https://github.com/stencila/thema/pull/111/files#diff-25876d902a7b9a110d8b13aa7d1d3af9R39

**Pull request merged**